### PR TITLE
v1.1.0

### PR DESCRIPTION
**Pull request by finitearth**

### CODE REVIEW

File: promptolution/tasks/classification_tasks.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
 """Module for classification tasks."""
 
+import json
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:3]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 import json
 from pathlib import Path
 from typing import Callable, Dict, List, Literal, Optional
+from typing import Callable, Dict, List, Literal, Optional
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:7]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 3]
```diff
@@ -20,15 +23,18 @@ class ClassificationTask(BaseTask):
     Attributes:
         task_id (str): Unique identifier for the task.
         path (Path): Path to the dataset description JSON file, and initial prompts.
+        path (Path): Path to the dataset description JSON file, and initial prompts.
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:26]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 4]
```diff
@@ -20,15 +23,18 @@ class ClassificationTask(BaseTask):
...
         dataset_json (Dict): Dictionary containing dataset information.
         description (Optional[str]): Description of the task.
         initial_population (Optional[List[str]]): Initial set of prompts.
         xs (Optional[np.ndarray]): Input data for the task.
         ys (Optional[np.ndarray]): Ground truth labels for the task.
         classes (Optional[List]): List of possible class labels.
         seed (int): Random seed for reproducibility.
+        seed (int): Random seed for reproducibility.
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:34]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 5]
```diff
@@ -20,15 +23,18 @@ class ClassificationTask(BaseTask):
...
         split (Literal["dev", "test"]): Dataset split to use.
         metric (Callable): Metric to use as an evaluation score for the prompts.
+        metric (Callable): Metric to use as an evaluation score for the prompts.
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:37]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 6]
```diff
@@ -47,23 +53,84 @@ def __init__(
         Args:
             task_id (str): Unique identifier for the task.
             dataset_path (str): Path to the dataset description JSON file.
+            dataset_path (str): Path to the dataset description JSON file.
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:56]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 7]
```diff
@@ -47,23 +53,84 @@ def __init__(
...
             seed (int, optional): Random seed for reproducibility. Defaults to 42.
             split (Literal["dev", "test"], optional): Dataset split to use. Defaults to "dev".
             metric (Callable): Metric to use as an evaluation score for the prompts. Defaults to sklearn's accuracy.
+            metric (Callable): Metric to use as an evaluation score for the prompts. Defaults to sklearn's accuracy.
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:60]: duplicate

File: promptolution/tasks/classification_tasks.py [hunk 8]
```diff
@@ -47,23 +53,84 @@ def __init__(
...
         """
         self.task_id: str = task_id
         self.path: Path = dataset_path
         self.dataset_json: Dict = json.loads((dataset_path / Path("description.json")).read_text())
+        self.path: Path = dataset_path
+        self.dataset_json: Dict = json.loads((dataset_path / Path("description.json")).read_text())
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:66]: both lines duplicate

File: promptolution/tasks/classification_tasks.py [hunk 9]
```diff
@@ -143,6 +213,17 @@ def evaluate(
...
+
+        if return_seq:
+            preds, seqs = preds
+
+        scores = np.array([self.metric(ys_subsample, pred) for pred in preds])
+
+        if return_seq:
+            return scores, seqs
+
+        return scores
```
Comment (timo282) [inline promptolution/tasks/classification_tasks.py:226]: whole new code segment (216-226) duplicate

File: pyproject.toml [hunk 10]
```diff
@@ -1,12 +1,12 @@
...
 name = "promptolution"
-version = "1.0.1"
+version = "1.1.0"
 description = ""
 authors = ["Tom Zehle, Moritz Schlager, Timo Heiß"]
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "^3.11"
+python = "^3.9"
```
Comment (timo282) [inline pyproject.toml:9]: can we add this information to the release notes? "enable support for python versions from 3.9 onwards (previously 3.11)" or similar

**Review by timo282 — APPROVED** [review]
Remove duplicate code (see comments below) and run pre-commit - CI workflow should then also run as expected

**Review by mo374z — APPROVED** [review]

**Review by mo374z — APPROVED** [review]

**Pull request merged**